rest/users/read.tex

```diff
@@ -1,19 +1,90 @@
 \subsubsection{READ}
 \label{sec:rest:api:user:read}
 Es sind nur Anfragen mit der HTTP-GET-Methode für ein READ auf die Daten zugelassen.
-Die Route gibt für den eingeloggten Benutzer seine persönlichen Daten wieder.
+Der Pfad gibt für den eingeloggten Benutzer seine persönlichen Daten wieder.
 Die Antwort erfolgt mit dem JSON-Objekt \reflisting{lst:code:rest:api:user:read:ret}. 
 Die einzelnen Felder der Antwort werden in \reftabllec{tab:rest:api:user:read:ret} beschrieben.
 \input{json}
 \begin{lstlisting}[caption={JSON-Antwort für einen GET-Aufruf des Pfads /api/users},label={lst:code:rest:api:user:read:ret},frame=tlrb]
+[
 {
- id: "<STRING>",
- student-id: "<STRING>"
- name: "<STRING>",
- surename: "<STRING>",
- dateofbirth: "<DATE>",
- sex: "<ENUM>",
+	id: "<STRING>",
+	studentId: "<STRING>"
+ 	firstName: "<STRING>",
+ 	preferredName: "<STRING>",
+ 	lastName: "<STRING>",
+ 	dateOfBirth: "<DATE>",
+ 	sex: "<ENUM>",
+ 	schools: [
+ 	{
+ 		school: {
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		role: "<STRING>",
+ 		start: "<DATE>",
+ 		end: "<DATE>"
+ 	},
+ 	...
+ 	],
+ 	classes: [
+ 	{
+ 		id: "<STRING>",
+ 		displayName: "<STRING>",
+ 		summary: "<STRING>",
+ 		school: {
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		schoolYear: {
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		roleWithinClass: "<STRING>",
+ 		start: "<DATE>",
+ 		end: "<DATE>"
+ 	},
+ 	...
+	],
+	subjects: [
+	{
+		id: "<STRING>",
+		displayName: "<STRING>",
+		schoolSubject: [
+ 		{
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		...
+ 		],
+ 		school: {
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		schoolYear: {
+ 			id: "<STRING>",
+ 			displayName: "<STRING>"
+ 		},
+ 		roleWithinSubject: "<STRING>",
+ 		start: "<DATE>",
+ 		end: "<DATE>" 		
+	},
+	...
+	],
+	guardians: [
+	{
+		id: "<STRING>",
+		role: "<STRING>",
+		user: {
+			id: "<STRING>",
+			firstName: "<STRING>",
+			lastName: "<STRING>"		
+		}
+	},
+	...
+	]
 }
+]
```

Comment:
Frage, wie so wurde hier das Benutzer Objekt in eine Liste/Array die genau ein element hat gepackt?

Reply:
korrigiert